### ISSUE TITLE
Apply librashader to any game/executable

### ISSUE THREAD

**Issue by user18081972** [SnowflakePowered/librashader#15]
Would it be possible to have librashader just apply to any game? For example I would like to use some librashaders with FF7 steam edition, the PC port of the classic PS1 game.

Could it just take the final output of my monitor as its input to apply the shader to?

**Comment by chyyran** [comment]
It's possible but that is not in scope for librashader. You would have to hook the swapchain of the game, fetch the rendered frame, and draw it back to the swapchain after applying the shader with librashader.

**Comment by cruduxcru0** [comment]
That would be a job for a ReShade fork. It already does an excellent job of hooking the needed render targets, swapchains, etc. All that needs to be gutted is its custom shader compiler (yes, Reshade has its own dedicated/homebrew shader compiler/lexer/SPIRV emitter/etc)

**Comment by chyyran** [comment]
Likely don’t even need a ReShade fork, can likely be done with a ReShade addon that just selects for the correct runtime and draws before present.

**Comment by chyyran** [comment]
Yeah it would be possible to implement this as a vulkan layer, but that would not be something to be done at the librashader level, but at a layer above by something that consumes librashader as a crate. 

I have a POC of a Vulkan layer in Rust at [snowflake-ingame](https://github.com/SnowflakePowered/snowflake-ingame/tree/master/snowflake-ingame/src/vk).

**Comment by ChthonVII** [comment]
I strongly second Quackdoc's idea to incorporate with gamescope. That would pretty much be the be-all, end-all for linux retro gaming.

**Comment by chyyran** [comment]
While I would love to see librashader in gamescope, that's up to Valve if they ever want to incorporate it. The header libraries are license-compatible with BSD, on Linux dynamic linking is the default anyways.
